app/controllers/posts_controller.rb

```diff
@@ -3,13 +3,18 @@ class PostsController < ApplicationController
   before_action :authenticate_user!
   before_action :correct_user, only: [:edit, :update, :destroy]
 
+  require 'will_paginate/array'
+
   # GET /posts
   # GET /posts.json
   def index
-    @posts = Post.all.order('updated_at DESC')
-
+    @posts = Post.where("user_id IN (?) OR user_id = ?", current_user.confirmed_friends_ids, current_user.id).order('updated_at DESC').paginate(page: params[:page], per_page: 10)
```

Comment:
Nice way to get the feed of an User. Another alternative would've been to just call `@current_user.friends.posts` but you would need to figure out the logic for it. [**OPTIONAL**]